src/TConsole.cpp

```diff
@@ -772,14 +777,31 @@ void TConsole::closeEvent( QCloseEvent *event )
 
     if( profile_name != "default_host" && ! mUserAgreedToCloseConsole )
     {
-        ASK: int choice = QMessageBox::question( this, "Exiting Session: Question", "Do you want to save the profile "+profile_name, QMessageBox::Yes|QMessageBox::No|QMessageBox::Cancel );
+        QCheckBox *cb = new QCheckBox( tr("Save the Window Layout before closing this profile.") );
+        QMessageBox *closePrompt = new QMessageBox( QMessageBox::Icon::Question, tr("Exiting Session: Question"), tr("Do you want to save the profile: %1").arg(profile_name), QMessageBox::Yes|QMessageBox::No|QMessageBox::Cancel, this );
+        closePrompt->setCheckBox(cb);
+
+        connect(cb, &QCheckBox::stateChanged, [this](int state){
+                if (static_cast<Qt::CheckState>(state) == Qt::CheckState::Checked) {
+                    this->mSaveLayoutRequested = true;
+                }
+            });
+
+        //ASK: int choice = QMessageBox::question( this, "Exiting Session: Question", "Do you want to save the profile "+profile_name, QMessageBox::Yes|QMessageBox::No|QMessageBox::Cancel );
+        ASK: mSaveLayoutRequested = false;
+        closePrompt->exec();
+        int choice = closePrompt->result();
```

Comment:
IIRC this and the line above and much of below _might_ be do-able as:

    switch (closePrompt->exec()) {
    case QMessageBox::Cancel :
        // stuff
        event->ignore();
        return:
    case QMessageBox::Yes : 
        // more stuff
        event->accept();
        return;
    case QMessageBox::No : 
        // and so on!


OTOH the presence of what I was always taught in __C__  was an _evil abomination_ :open_mouth: the `goto` keyword may prevent such things...